app/assets/javascripts/discourse/app/routes/application.js

```diff
@@ -25,6 +25,16 @@ function unlessReadOnly(method, message) {
   };
 }
 
+function unlessStrictlyReadOnly(method, message) {
+  return function () {
+    if (this.site.get("isReadOnly") && !this.site.get("isStaffWritesOnly")) {
```

Comment:
Would be better to use `this.site?.isReadonly`  or just `this.site.isReadonly` if site cannot be null/undefined